test.js

```diff
@@ -1538,6 +1538,29 @@ function runTests(baseopts) {
             done();
           });
       });
+      it('should emit an unlink event when a file is updated and deleted just after that', function(done) {
+        var spy = sinon.spy();
+        var testPath = getFixturePath('subdir/add.txt');
+        var filename = sysPath.basename(testPath);
+        options.cwd = sysPath.dirname(testPath);
+        fs.mkdir(options.cwd, w(function() {
+          fs.writeFile(testPath, 'hello', w(function() {
+            stdWatcher()
+              .on('all', spy)
+              .on('ready', function() {
+                fs.writeFile(testPath, 'edit', simpleCb);
+                w(function() {
```

Comment:
Shouldn't this be moved to be the callback after the `writeFile` above? Also, overriding the timeout to 10ms makes detecting the lack of a `change` event not very meaningful. The default time, or anything that gives it some time to react while still within the awf stability threshold would be better.